### PR TITLE
In order to suppress some deprecations warning :

### PR DESCRIPTION
**Pull request by GLacourt**
Enforce type declarations.
Add strict types declaration.
Update php minimum requirement.
Add composer vulnerability package

### CODE REVIEW

File: src/Doctrine/DBAL/Types/PhoneNumberType.php [hunk 1]
```diff
@@ -18,6 +19,7 @@
 use libphonenumber\PhoneNumber;
 use libphonenumber\PhoneNumberFormat;
 use libphonenumber\PhoneNumberUtil;
+use function gettype;
```
Comment (jderusse) [inline src/Doctrine/DBAL/Types/PhoneNumberType.php:22]: the project don't import functions

File: composer.json [hunk 2]
```diff
@@ -12,12 +12,13 @@
...
     },
     "require": {
-        "php": ">=7.4",
+        "php": ">=8.0",
         "giggsey/libphonenumber-for-php": "^8.0",
         "symfony/framework-bundle": "^4.4|^5.3|^6.0",
         "symfony/intl": "^4.4|^5.3|^6.0"
     },
     "require-dev": {
+        "roave/security-advisories": "dev-latest",
```
Comment (jderusse) [inline composer.json:21]: please revert

**Review by jderusse — CHANGES_REQUESTED** [review]

File: src/Doctrine/DBAL/Types/PhoneNumberType.php [hunk 3]
```diff
@@ -1,4 +1,5 @@
 <?php
+declare(strict_types=1);
```
Comment (Nek-) [inline src/Doctrine/DBAL/Types/PhoneNumberType.php:2]: This may be a regression as well

**Comment by Nek-** [comment]
Hello @GLacourt and thank you for your contribution. However you was not answering to our comments so I took the liberty to make another PR containing your changes but with a scope at the project level. https://github.com/odolbeau/phone-number-bundle/pull/120

Let me know if something looks wrong to you inside this.

**Pull request closed without merging**